main.cpp

```diff
@@ -31,6 +31,11 @@
 #pragma warning(disable : 4996)
 #endif
 
+#ifdef __clang__
+#pragma clang diagnostic ignored "-Wswitch-enum"
```

Comment:
Is it necessary for these to be here? I assume they would only be relevant if you intentionally add -Weverything, but it is not applied in the build system.